TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/AssetController.php

```diff
@@ -67,6 +98,43 @@ public function initializeObject()
     }
 
     /**
+     * Edit an asset
+     *
+     * @param Asset $asset
+     * @return void
+     */
+    public function editAction(Asset $asset)
+    {
+        $context = $this->createContext();
+        $relatedDocumentNodes = [];
+        foreach ($this->getRelatedNodes($asset) as $relatedNodeData) {
+            $node = $this->nodeFactory->createFromNodeData($relatedNodeData, $context);
+            $flowQuery = new FlowQuery(array($node));
+            /** @var Node $documentNode */
+            $documentNode = $flowQuery->closest('[instanceof TYPO3.Neos:Document]')->get(0);
+            $relatedDocumentNodes[$documentNode->getIdentifier()] = $documentNode;
+        }
+
+        $this->view->assignMultiple(array(
+            'tags' => $asset->getAssetCollections()->count() > 0 ? $this->tagRepository->findByAssetCollections($asset->getAssetCollections()->toArray()) : $this->tagRepository->findAll(),
+            'asset' => $asset,
+            'assetCollections' => $this->assetCollectionRepository->findAll(),
+            'relatedDocumentNodes' => $relatedDocumentNodes
+        ));
+    }
+
+    /**
+     * @return \TYPO3\TYPO3CR\Domain\Service\Context
+     */
+    protected function createContext()
+    {
+        $account = $this->securityContext->getAccount();
+        /** @var Workspace $personalWorkspace */
+        $personalWorkspace = $this->workspaceRepository->findOneByName(User::getPersonalWorkspaceNameForUsername($account->getAccountIdentifier()));
+        return $this->createContentContext($personalWorkspace->getName());
```

Comment:
it's not good using the security context and these methods here.. instead use the UserService->getPersonalWorkspaceName to get the personal workspace name